src/settings/LoanHistory/LoanHistoryForm.js

```diff
@@ -0,0 +1,172 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  injectIntl,
+  intlShape,
+  FormattedMessage,
+} from 'react-intl';
+import {
+  Field,
+  FieldArray,
+} from 'redux-form';
+
+import stripesForm from '@folio/stripes/form';
+import {
+  Button,
+  Checkbox,
+  Col,
+  Pane,
+  Row,
+  Headline,
+  RadioButton,
+} from '@folio/stripes/components';
+
+import { Period } from '../components';
+import optionsGenarator from '../utils/options-genarator';
+import {
+  closingTypes,
+  closingTypesMap,
+  intervalPeriods,
+} from '../../constants';
+
+import css from './LoanHistoryForm.css';
+
+class LoanHistoryForm extends Component {
+  constructor(props) {
+    super(props);
+    this.genarateOptions = optionsGenarator.bind(null, props.intl.formatMessage);
+    this.selectedPeriods = intervalPeriods.filter(period => props.initialValues.selectedPeriodsValues.includes(period.value));
+  }
+
+  onSave = data => this.props.onSubmit({ loan_history: JSON.stringify(data) });
+
+  getLastMenu = () => {
+    const { pristine, submitting } = this.props;
+
+    return (
+      <Button
+        type="submit"
+        disabled={pristine || submitting}
+        id="clickable-loan-history-save-button"
+        marginBottom0
+      >
+        <FormattedMessage id="ui-circulation.settings.loanHistory.save" />
+      </Button>
+    );
+  }
+
+  renderPeriod = () => {
```

Comment:
```suggestion
  renderPeriod() {
```
Use arrow syntax only when really needed.